addons/mail/static/src/models/message_view.js

```diff
@@ -565,5 +597,22 @@ registerModel({
             },
             inverse: 'messageViewOwner',
         }),
+        /**
+         * Sender (impersonation note)
+         */
+        senderView: attr({
+            compute() {
+                if (this.messaging.currentGuest) {
```

Comment:
Being guest is not exclusive with being logged in (cookie is preserved afterwards).

But the condition is probably not necessary?